src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -714,21 +687,44 @@ uct_cuda_copy_md_query_attributes(uct_cuda_copy_md_t *md, const void *address,
         return UCS_ERR_NO_DEVICE;
     }
 
-    uct_cuda_copy_sync_memops(md, address, is_vmm);
+    if (cuda_mem_ctx == NULL) {
+        status = uct_cuda_copy_md_primary_ctx_retain(cuda_device,
+                                                     &cuda_mem_ctx);
+        if (status != UCS_OK) {
+            return status;
+        }
+
+        primary_ctx_retained = 1;
+    }
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxPushCurrent(cuda_mem_ctx));
```

Comment:
what if it is already current context? maybe no nee to push then?

Reply:
It was done in the same was as in previous PR (https://github.com/openucx/ucx/pull/10388) to simplify the code.